library/FeeSheet.class.php

```diff
@@ -925,28 +927,49 @@ public function save(
             $main_supid = 0;
         }
 
+        function formatICD10Codes($inputArray): string
+        {
+            $result = [];
+
+            foreach ($inputArray as $item) {
+                // Check if the code_type is ICD10 and the code is set
+                if (isset($item['code_type'], $item['code']) && $item['code_type'] === 'ICD10') {
+                    $result[] = 'ICD10|' . $item['code'];
+                }
+            }
+
+            // Join the formatted codes with a comma
+            return implode(',', $result);
+        }
+
         $copay_update = false;
         $update_session_id = '';
         $ct0  = ''; // takes the code type of the first fee type code type entry from the fee sheet, against which the copay is posted
         $cod0 = ''; // takes the code of the first fee type code type entry from the fee sheet, against which the copay is posted
         $mod0 = ''; // takes the modifier of the first fee type code type entry from the fee sheet, against which the copay is posted
 
+        $justifyCpt = formatICD10Codes($bill);
+
         if (is_array($bill)) {
             foreach ($bill as $iter) {
                         // Skip disabled (billed) line items.
                 if (!empty($iter['billed'])) {
                     continue;
                 }
-
+                if ($iter['code_type'] === 'CPT4' && empty($iter['justify'])) {
+                    $j = $justifyCpt;
+                } else {
+                    unset($j);
+                }
                 $id        = $iter['id'] ?? null;
                 $code_type = $iter['code_type'];
                 $code      = $iter['code'];
                 $del       = !empty($iter['del']);
                 $units     = empty($iter['units']) ? 1 : intval($iter['units']);
                 $pricelevel = empty($iter['pricelevel']) ? '' : $iter['pricelevel'];
                 $revenue_code  = empty($iter['revenue_code']) ? '' : trim($iter['revenue_code']);
-                $modifier  = empty($iter['mod']) ? '' : trim($iter['mod']);
-                $justify   = empty($iter['justify'  ]) ? '' : trim($iter['justify']);
+                $modifier  = empty($iter['mod']) ? trim($j) : trim($iter['mod']);
+                $justify   = empty($iter['justify'  ]) ? trim($this) : trim($iter['justify']);
```

Comment:
hi @juggernautsei , this looks like a typo

Reply:
@stephenwaite that is not the latest change. I will review